src/js/components/stock-movement-wizard/request/CreateStockMovement.jsx

```diff
@@ -198,6 +231,14 @@ class CreateStockMovement extends Component {
     if (this.state.values.origin && this.state.values.destination) {
       this.fetchStockLists(this.state.values.origin, this.state.values.destination);
     }
+    if (this.state.values.origin) {
+      if (this.state.values.origin?.supportedActivities?.includes(ActivityCode.APPROVE_REQUEST)) {
+        this.setSupportsApprover(true);
+        this.fetchAvailableApprovers();
+      } else {
+        this.setSupportsApprover(false);
+      }
```

Comment:
What is the default value of `supportsApprover`? Maybe if it is false it could be shortened to
```
      if (this.state.values.origin?.supportedActivities?.includes(ActivityCode.APPROVE_REQUEST)) {
        this.setSupportsApprover(true);
        this.fetchAvailableApprovers();
      }
```
(if it works I can see 2 more similar places to change)

Reply:
@alannadolny you can change the origin many times, so you could potentially set an origin that doesn't support the request approval and the flag would not change from `true` to `false`

Reply:
the default value is `false`, but this `this.setSupportsApprover(false);` is needed in cases when we are changing the selected fulfilling location from  **supports APPROVE_REQUEST** to  **not supports APPROVE_REQUEST**.

If my explanation is a bit confusing, let me give you an example.
I chose a location that supports APPROVE_REQUEST  -> the select should appear
then I change the location which does not support APPROVE_REQUEST  -> the select should disappear